designs/osc_high_availability/OSC_High_Availability.md

```diff
@@ -0,0 +1,83 @@
+# OSC High Availability
+- OSC High Availability eliminates from single points of failure which protects from unplanned downtime.
+- Performs regular checks to make sure each OSC Service Node is working properly, and if one fails, switch it out for one that is working.
+- It tries to bring back the failed services into the cluster and maintains the same number of services node.
+    
+#### There are 2 types of failures
+1. Service Stops
+	OSC service may be stopped abnormaly.
+2. Node Stops
+	OSC Service Node Stopped because of some reasons.
+
+## Background
+Currently, OSC does not have the HA configuration option to handle above failure types.So,A high availability OSC will work by having more service nodes than it needs,will perform regular checks to make sure each service/node is working properly, and if one fails, switching it out for one that is working.
+
+We have explored two approaches.
+1. Two nodes cluster with Active/Passive setup using Pacemaker,Corosync and Distributed Replicated Block Device(DRBD).
+2. Segregate Application and Persistence Layers.
+
+## OSC HA with Active/Passive Setup 
+Two nodes cluster with Active/Passive setup using Pacemaker,Corosync and Distributed Replicated Block Device(DRBD).
+Active Node serve the requests where as Passive Node is Standby.So,on fails of Active Node,Passive Node will get promoted to Active Node.
```

Comment:
General Comment: Space after ".". Fix other instances of this too in the document(if any).